src/Pagerfanta/Pagerfanta.php

```diff
@@ -498,4 +499,24 @@ private function needsToIntegerConversion($value)
     {
         return (is_string($value) || is_float($value)) && (int) $value == $value;
     }
+
+    /**
+     * Get page number of the item at specified position (1-based index)
+     *
+     * @param integer $position
+     *
+     * @return integer
+     */
+    public function getPageNumberForItemAtPosition($position)
+    {
+        if (!is_int($position)) {
+            throw new NotIntegerItemException();
+        }
+
+        if ($this->getNbResults() < $position) {
+            throw new LogicException('Searched item does not exist');
```

Comment:
I think that this should be an `OutOfBoundsException`.

Also, your exception message could be phrased better.  I recommend this:

```
throw new OutOfBoundsException(sprintf(
    'Item requested at position %d, but there are only %d items.',
    $position,
    $this->getNbResults()
));
```